include/zephyr/bluetooth/conn.h

```diff
@@ -1358,6 +1358,22 @@ struct bt_conn_auth_info_cb {
  */
 int bt_conn_auth_cb_register(const struct bt_conn_auth_cb *cb);
 
+/** @brief Register authentication callbacks used only for a given connection.
+ *
+ *  Register callbacks to handle authenticated pairing for a given connection.
+ *  The set of callbacks overlays globally defined callbacks.
+ *
+ *  This function can be used only for Bluetooth LE connections.
+ *  The authentication callbacks for a given connection cannot be updated if security procedures in
+ *  the SMP module have already started. This function can be called only once for connection.
+ *
+ *  @param conn	Connection object.
+ *  @param cb	Callback struct.
+ *
+ *  @return Zero on success or negative error code otherwise
+ */
+int bt_conn_auth_cb_register_conn(struct bt_conn *conn, const struct bt_conn_auth_cb *cb);
```

Comment:
That would cause naming issue with the existing `bt_conn_auth_cb_register`.

Reply:
Ah, I forgot that we had prefixed the original callbacks also with `bt_conn` (in hindsight, simply `bt_auth_cb` might have been better).

Reply:
I still think that we can come up with a more descriptive suffix than "conn". Just brainstorming, but how about `bt_conn_auth_cb_override()` or `bt_conn_auth_cb_dedicated()` (not really a fan of the last one though, since it lacks a verb). Thoughts?

Reply:
I also think `bt_conn_auth_cb_override()` could be better than used `conn` suffix (the argument list clearly suggests that it's used only for a given conn). We could consider `bt_conn_auth_cb_overlay()` too maybe (since we actually overlay the callbacks for given connection - global callbacks are still there)?

I do not have strong preference between these two. As soon as we agree on used name I will push the update.

Reply:
@jhedberg, @alwa-nordic, do you have preference regarding naming here? `bt_conn_auth_cb_override()`,  `bt_conn_auth_cb_overlay()` or maybe you have another proposition?

Reply:
Maybe `bt_conn_auth_cb_register_per_conn()`?

Reply:
`per_con` sounds (at least to me) like a general statement for multiple connections, when this is about one specific connection. `override` and `overlay` are both fine with me. I'll let @MarekPieta pick one :)

Reply:
Changed to overlay. Thanks for the comments.